### PR TITLE
Send back error message and status code if there's an error in the pulls Hook

### PR DESCRIPTION
**Pull request by wilsonge**
Part of the fix to #681 

Note if there is an exception now when performing the hook action we no longer close the application at that point but return an error message an appropriate status header

### CODE REVIEW

File: src/App/Tracker/Controller/Hooks/ReceivePullsHook.php [hunk 1]
```diff
@@ -61,11 +61,16 @@ protected function prepareResponse()
...
 		}
 		else
 		{
-			$this->insertData();
+			$result = $this->insertData();
+		}
+
+		if (!$result)
+		{
+			$this->getContainer()->get('app')->close();
```
Comment (mbabker) [inline src/App/Tracker/Controller/Hooks/ReceivePullsHook.php]: The parent controller deals with the close.  IMO this should be a different message, not just a blank return (look at the screen cap I attached earlier).

File: src/JTracker/Controller/AbstractAjaxController.php [hunk 2]
```diff
@@ -79,6 +79,20 @@ public function execute()
...
 	/**
+	 * Allows setting the status header into the application
+	 *
+	 * @param   int  $code  The status code to set into the application
+	 *
+	 * @return  void
+	 *
+	 * @since   __DEPLOY_VERSION__
+	 */
+	protected function setHeader($code = 200)
```
Comment (mbabker) [inline src/JTracker/Controller/AbstractAjaxController.php]: Bad method name IMO, implying you can set all the app headers when you've coded only support for the status header.

Comment (wilsonge) [inline src/JTracker/Controller/AbstractAjaxController.php]: I agree - not sure why I did that - fixed

File: src/App/Tracker/Controller/Hooks/ReceivePullsHook.php (hunk 1 above)
Comment (wilsonge) [inline src/App/Tracker/Controller/Hooks/ReceivePullsHook.php]: The close was something that was already done inside the respective functions. But yah I've now let things continue and set the error message to be the same as the log message

**Pull request merged**